### ISSUE TITLE
could not find type std::AgentConfig in namespace std

### ISSUE THREAD

**Issue by automate-solutions** [inmanta/openstack#4]
When using the latest version of the openstack module (or any 2.2.X version), I got this error when exporting:
```
[root@inmanta openstack-test]# inmanta -vvv export
INFO     Creating new virtual environment in ./.env
DEBUG    Starting compile
INFO     verifying project
WARNING  Empty deployment model.
could not find type std::AgentConfig in namespace std (reported in Construct(std::AgentConfig) (/tmp/tmpmqtlmkgr/openstack/model/_init.cf:42))
```
Shouldn't the compiler version be raised to 2017.2 like in the std module?
`compiler_version: 2017.2`

**Comment by bartv** [comment]
The latest openstack versions use this from std. The problem is that the std module contains features that require a newer compiler. Hence, the error. We are getting close to a new release so you could try to upgrade to the latest dev version:

https://pkg.inmanta.com/inmanta-oss-dev/

**Comment by automate-solutions** [comment]
Sure. I'm gonna try and install the dev version.
Nevertheless, shouldn't it be safer to enforce the new version of the compiler in module.yml in order to avoid those kind of cross dependencies' issues?

**Comment by bartv** [comment]
The problem with this issue is that it is not the openstack module that requires a newer compiler. We should rethink the way we release new module versions.

I added an issue to the main compiler to change the way we release new module versions. inmanta/inmanta#442

**Comment by bartv** [comment]
This was fixed in version 2.3.3 (just released)

**Comment by bartv** [comment]
My previous comment was for an other issue. However, the issue causing this should be handled in the core with an improved branching model.
